pkg/api/chunk_test.go

```diff
@@ -118,6 +120,29 @@ func TestChunkUploadDownload(t *testing.T) {
 		}
 
 	})
+	t.Run("retrieve-targets", func(t *testing.T) {
+		jsonhttptest.ResponseDirect(t, client, http.MethodPost, resource(validHash), bytes.NewReader(validContent), http.StatusOK, jsonhttp.StatusResponse{
+			Message: http.StatusText(http.StatusOK),
+			Code:    http.StatusOK,
+		})
+
+		// try to fetch the same chunk
+		resp := request(t, client, http.MethodGet, resourceTargets(validHash), nil, http.StatusOK)
+		data, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if !bytes.Equal(validContent, data) {
+			t.Fatal("data retrieved doesnt match uploaded content")
+		}
+
+		// Check if the target is obtained correctly
+		if resp.Header.Get("Targets") != targets {
+			t.Fatal("Invalid Targets")
+		}
+
+	})
```

Comment:
do we really need to:
1. upload a chunk
2. download it
3. verify its data

just to check that targets are obtained correctly?

Reply:
are you sure we need to? 

please see my suggestion in the review down below, just my 2¢